### ISSUE TITLE
Different format  in HostResponse and HostForMigrationResponse

### ISSUE THREAD

**Issue by weizhouapache** [apache/cloudstack#4460]
The value of memoryallocated in the output of listhosts and findhostformigration are different.
```
(local) mgt01 > list hosts filter=cpuallocated,memoryallocated type=Routing
{
  "count": 5,
  "host": [
    {
      "cpuallocated": "12.51%",
      "memoryallocated": 1879048192
    },
    {
      "cpuallocated": "1.14%",
      "memoryallocated": 536870912
    },
    {
      "cpuallocated": "0%",
      "memoryallocated": 0
    },
    {
      "cpuallocated": "0%",
      "memoryallocated": 0
    },
    {
      "cpuallocated": "18.19%",
      "memoryallocated": 7247757312
    }
  ]
}
```

find host for migration
```
(local) mgt01 > find hostsformigration virtualmachineid=39 filter=cpuallocated,memoryallocated
{
  "count": 5,
  "host": [
    {
      "cpuallocated": "0%",
      "memoryallocated": "0%"
    },
    {
      "cpuallocated": "0%",
      "memoryallocated": "0%"
    },
    {
      "cpuallocated": "1.14%",
      "memoryallocated": "7.36%"
    },
    {
      "cpuallocated": "12.73%",
      "memoryallocated": "29.43%"
    }
  ]
}

```

they should be same.

##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

~~~

~~~

##### CLOUDSTACK VERSION

~~~
4.14
~~~

##### STEPS TO REPRODUCE

~~~

~~~

##### EXPECTED RESULTS

~~~
memoryallocated in two responses are in same format (amount or percentage)
~~~

##### ACTUAL RESULTS

~~~

~~~

**Comment by weizhouapache** [comment]
this is regression of commit 09de5e0b51d1cd0e1d84eef120ae48358300fdcd

**Comment by DaanHoogland** [comment]
fixed with #4478
